orc/src/main/java/org/apache/iceberg/orc/OrcSchemaWithTypeVisitor.java

```diff
@@ -71,12 +71,12 @@ protected T visitRecord(
     return visitor.record(struct, record, names, results);
   }
 
-  private static <T> T visitUnion(Type type, TypeDescription union, OrcSchemaWithTypeVisitor<T> visitor) {
+  protected T visitUnion(Type type, TypeDescription union, OrcSchemaWithTypeVisitor<T> visitor) {
```

Comment:
+1 on why this change

Reply:
This is just to align with the same code pattern of other visitor's method, see `visitRecord`, also, this change enables subclass overriding, which might be necessary in the future, since this `OrcSchemaWithTypeVisitor` is a very generic visitor.